graph/src/util/futures.rs

```diff
@@ -52,21 +53,34 @@ where
         let logger = logger.clone();
 
         attempt_count += 1;
-        let delay_ms = (1 << attempt_count).min(max_delay_ms);
 
-        try_it().map_err(move |e| {
-            if attempt_count >= log_after {
-                debug!(
-                    logger,
-                    "Trying again in ~{} seconds after {} failed (attempt #{})",
-                    (delay_ms as f64 / 1000.0).round(),
-                    &operation_name,
-                    attempt_count + 1,
-                );
-            }
+        try_it()
+            .deadline(Instant::now() + Duration::from_secs(60))
```

Comment:
60 seconds is twice as long as most network timeouts in other systems (which are usually 30s). I'd even make it less in the hopes of speeding things up. 10s? 20s? But at most 30s.

Reply:
Hmmm I did some testing with 30s and it actually ends up preventing some calls from completing. In particular, I saw eth_getLogs consistently take over 40 seconds... it's long but I think 60s would be safer

Reply:
Ok, that's fair enough. Better than nothing!